File: rdkit-sys/wrapper/src/frag_catalog.cc
#include "rust/cxx.h"
#include <GraphMol/FragCatalog/FragCatParams.h>
// #include <GraphMol/GraphMol.h>
#include <GraphMol/MolOps.h>
#include <memory>

namespace RDKit {
std::shared_ptr<FragCatParams> init_frat_cat_from_string(const std::string &text) {
	return std::shared_ptr<FragCatParams>(new FragCatParams(text));
}

// here we need to do something
} // namespace RDKit

File: rdkit-sys/src/bridge/frag_catalog.rs
#[cxx::bridge(namespace = "RDKit")]
pub mod ffi {
    unsafe extern "C++" {
        include!("wrapper/include/frag_catalog.h");

        pub type FragCatParams;

        pub fn init_frat_cat_from_string(text: &CxxString) -> SharedPtr<FragCatParams>;
    }
}
